src/main/java/com/scalyr/util/Util.java

```diff
@@ -0,0 +1,52 @@
+package com.scalyr.util;
+
+import com.scalyr.api.logs.EventAttributes;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public final class Util {
+  public static Integer stringToIntMemory(String maxBufferRam) {
+    if(maxBufferRam != null && !"".equals(maxBufferRam)) {
+      maxBufferRam = maxBufferRam.toLowerCase().trim();
+      if(maxBufferRam.contains("m")) {
+        return Integer.valueOf(maxBufferRam.substring(0, maxBufferRam.indexOf("m"))) * 1048576;
+      } else if (maxBufferRam.contains("k")) {
+        return Integer.valueOf(maxBufferRam.substring(0, maxBufferRam.indexOf("k"))) * 1024;
+      } else {
+        return Integer.valueOf(maxBufferRam);
+      }
+    }
+    return null;
+  }
+
+  /**
+   * Mutation feels dirty...
```

Comment:
How about replacing this line with:

Mutate the supplied EventAttributes, by adding an entry for each key=value pair in extraAttributes.

Reply:
replaced this with a function that returns an EventAttributes with the kv stuff from the string. then we do EventAttributes#addAll(EventAttributes) to add the KV events to the other ones. no more mutation.